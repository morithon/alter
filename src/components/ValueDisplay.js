import React from 'react';
import {Text, View, StyleSheet, TouchableHighlight} from 'react-native';
import GAME_STATES from './gameStates';

const styles = StyleSheet.create({
	container: {
		flex: 20,
		backgroundColor: 'skyblue',
		alignItems: 'stretch',
		justifyContent: 'center'
	},
	text: {
		fontSize: 42,
		fontWeight: 'bold'
	},
	wordContainer: {
		flex: 1,
		alignItems: 'center',
		justifyContent: 'center'
	}
});

type ValueDisplayProps = {
	mode: 'displayValues' | 'waitForTouch';
	value: string;
	focusOn: boolean;
	onPress: Function;
};

const ValueDisplay = ({mode, value, focusOn, onPress}: ValueDisplayProps) => (
	<View style={styles.container}>
		{mode === GAME_STATES.DisplayValues && (
			<View style={styles.wordContainer}>
				<Text style={styles.text}>{value}</Text>
			</View>
		)}
		{mode === GAME_STATES.WaitForUserPress && (
			<TouchableHighlight
				style={styles.wordContainer}
				onPress={() => onPress(focusOn)}
			>
				<Text style={styles.text}>
					{focusOn && '+'}
				</Text>
			</TouchableHighlight>
		)}
	</View>
);

export default ValueDisplay;