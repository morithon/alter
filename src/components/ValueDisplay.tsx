import React from 'react';
import {Text, View, StyleSheet, TouchableHighlight} from 'react-native';
import {GameStates} from './GameStates';
import Word from './Word';

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#4fc3f7',
	},
	text: {
		fontSize: 42,
		fontWeight: 'bold'
	},
	wordContainer: {
		flex: 1,
		alignItems: 'center',
		justifyContent: 'center'
	}
});

export interface ValueDisplayProps {
	mode: GameStates;
	value: string;
	focusOn: boolean;
	onPress: (focusOn: boolean) => null;
	onWordFadeOut?: () => any;
};

const ValueDisplay = ({mode, value, focusOn, onPress, onWordFadeOut}: ValueDisplayProps) => (
	<View style={styles.container}>
		{mode === GameStates.DISPLAY_VALUES && (
			<View style={styles.wordContainer}>
				<Word
					value={value}
					fadeToDuration={50}
					showForDuration={400}
					onFadeOut={onWordFadeOut}
				/>
			</View>
		)}
		{mode === GameStates.WAIT_FOR_USER_PRESS && (
			<TouchableHighlight
				style={styles.wordContainer}
				onPress={() => onPress(focusOn)}
			>
				<Text style={styles.text}>
					{focusOn && '+'}
				</Text>
			</TouchableHighlight>
		)}
	</View>
);

export default ValueDisplay;