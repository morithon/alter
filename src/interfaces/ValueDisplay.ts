import { GameStates } from '../components/GameStates';

export interface ValueDisplayProps {
	mode: GameStates;
	value: string;
	focusOn: boolean;
	onPress: (focusOn: boolean) => void;
	onWordFadeOut?: () => void;
}
