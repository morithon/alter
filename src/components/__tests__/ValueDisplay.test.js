import React from 'react';
import ValueDisplay from '../ValueDisplay';

import renderer from 'react-test-renderer';

it('render matches snapshot', () => {
  expect(renderer.create(<ValueDisplay mode={'displayValues'} value={'testValue'}/>)).toMatchSnapshot();
  expect(renderer.create(<ValueDisplay mode={'waitForTouch'} focusOn={true}/>)).toMatchSnapshot();
  expect(renderer.create(<ValueDisplay mode={'waitForTouch'} focusOn={false}/>)).toMatchSnapshot();
});