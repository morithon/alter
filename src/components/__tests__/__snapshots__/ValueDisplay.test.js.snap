// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "skyblue",
      "flex": 20,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 42,
          "fontWeight": "bold",
        }
      }
    >
      testValue
    </Text>
  </View>
</View>
`;

exports[`render matches snapshot 2`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "skyblue",
      "flex": 20,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 42,
          "fontWeight": "bold",
        }
      }
    >
      +
    </Text>
  </View>
</View>
`;

exports[`render matches snapshot 3`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "skyblue",
      "flex": 20,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 42,
          "fontWeight": "bold",
        }
      }
    />
  </View>
</View>
`;
